### DISCUSSION TITLE
Is continuous app development and deployment possible?

### DISCUSSION THREAD

**Discussion by FloMiLe** [Budibase/budibase#1145] [Help] [answered]
I am unclear as to whether it is possible to create a web app with BB, deploy it, and then extend its features and upgrade the deployed app again.

Does that work? Are data then also updated in the local BB Builder? How will conflicts like deleted columns be handled?

Or is this just not possible, meaning that one has to develop the feature-complete app before deploying it?

**Comment by mike12345567** [comment, marked as answer]
Hi @FloMiLe - this is currently possible, if you deploy an app again some time after originally deploying an app it will work - the way that conflicts/merges are handled is via [CouchDB's replication strategy](https://guide.couchdb.org/draft/replication.html#:~:text=Replication%20is%20a%20one%2Doff,the%20source%20to%20the%20target.&text=This%20call%20sends%20all%20the,%3A%2F%2Fexample.org%2Fdatabase%20.) which will compare your local database to the existing one and then merge them. This means that any data which was created in your deployed app will be maintained, issues like column deletion will be fine as this will update the schema of the table in the live app.
There are still some conflict issues that we have yet to solve, for example deploying a user account which you've been testing with locally and have added relationships to, if the user has changed significantly in your local builder to the version in the deployed app then there could be conflicts. For now we recommend testing with a user account that doesn't get used in the deployed app to avoid issues.
